pyaedt/filtersolutions_core/optimization_goals_table.py

```diff
@@ -0,0 +1,411 @@
+
+    - RATIO: Represents transmission zeros ratio table.
+    - BANDWIDTH: Represents transmission zeros bandwidth table.
+    - LOWER_FREQUENCY = Represents lower frequency parameters.
+    - UPPER_FREQUENCY = Represents upper frequency parameters.
+    - GOAL_VALUE = Represents goal value parameters.
+    - CONDITION = Represents condition parameters.
+    - PARAMETER_NAME = Represents name of parameters.
+    - WEIGHT = Represents weight parameters.
+     -ENABLED = Represents status of using the goal parameters.
+    """
+
+    LOWER_FREQUENCY = 0
+    UPPER_FREQUENCY = 1
+    GOAL_VALUE = 2
+    CONDITION = 3
+    PARAMETER_NAME = 4
+    WEIGHT = 5
+    ENABLED = 6
+
+
+class OptimizationGoalsTable:
+    """Manipulates access to the entries of optimization goals table.
+
+    This class lets you to enter, edit or remove goal parameters in the optimization goals table.
+    The table includes the lower frequency, upper frequency, goal desired value, goal condition,
+    parameter name, weight, and enabled status.
+    """
+
+    def __init__(self):
+        self._dll = pyaedt.filtersolutions_core._dll_interface()._dll
+        self._dll_interface = pyaedt.filtersolutions_core._dll_interface()
+        self._define_multiple_bands_dll_functions()
+
+    def _define_multiple_bands_dll_functions(self):
+        """Define C++ API DLL functions."""
+        self._dll.getOptimizationGoalDefinitionRowCount.argtype = POINTER(c_int)
+        self._dll.getOptimizationGoalDefinitionRowCount.restype = c_int
+
+        self._dll.getOptimizationGoalDefinitionRow.argtype = [c_int, POINTER(c_char_p), c_int]
+        self._dll.getOptimizationGoalDefinitionRow.restype = c_int
+
+        self._dll.updateOptimizationGoalDefinitionRow.argtype = [c_int, c_int, c_char_p]
+        self._dll.updateOptimizationGoalDefinitionRow.restype = c_int
+
+        self._dll.appendOptimizationGoalDefinitionRow.argtype = [c_int, c_char_p]
+        self._dll.appendOptimizationGoalDefinitionRow.restype = c_int
+
+        self._dll.insertOptimizationGoalDefinitionRow.argtypes = [c_int, c_char_p, c_char_p]
+        self._dll.insertOptimizationGoalDefinitionRow.restype = c_int
+
+        self._dll.removeOptimizationGoalDefinitionRow.argtype = c_int
+        self._dll.removeOptimizationGoalDefinitionRow.restype = c_int
+
+        self._dll.adjustGoalFrequency.argtype = c_char_p
+        self._dll.adjustGoalFrequency.restype = c_int
+
+    @property
+    def row_count(self) -> int:
+        """Number of golas in the optimization goals table.
+        The default is `0`.
+
+        Returns
+        -------
+        int
+        """
+        table_row_count = c_int()
+        status = self._dll.getOptimizationGoalDefinitionRowCount(byref(table_row_count))
+        pyaedt.filtersolutions_core._dll_interface().raise_error(status)
+        return int(table_row_count.value)
+
+    def row(self, row_index) -> list:
+        """Get the row parameters of the optimization goals table as a list.
```

Comment:
```suggestion
        """Get the values for one row of the optimization goals table. The values are returned as a list: [value1, value2, value3].
```

The user will need to know what each entry in the list represents. Maybe something like the above but put in something descriptive for value1, value2, etc. There would be other ways to format it too if that format I proposed isn't neat and easy to read.